services/external/stripe/actions/delete-subscription.ts

```diff
@@ -0,0 +1,44 @@
+import { db } from "@/drizzle/db";
+import { subscriptions, teamMemberships } from "@/drizzle/schema";
+import { and, eq, ne } from "drizzle-orm";
+import { stripe } from "../config";
+
+export async function deleteSubscription(subscriptionId: string) {
+	const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+
+	if (subscription.status === "canceled") {
+		// Get the team_db_id from subscriptions table
+		const [sub] = await db
+			.select({ teamDbId: subscriptions.teamDbId })
+			.from(subscriptions)
+			.where(eq(subscriptions.id, subscriptionId))
+			.limit(1);
+
+		if (sub) {
```

Comment:
Could we handle these cases as exceptions? While these scenarios shouldn't occur in normal operation, I think we should throw exceptions to be notified if they do happen, rather than silently continuing:
- subscription not found
- earliestAdmin not found

This would help us detect any potential issues early. What do you think?

Reply:
updated c7a5f3ae0cb0 ♻️